src/journal/bsod.c

```diff
@@ -0,0 +1,209 @@
+/* SPDX-License-Identifier: LPGL-2.1-or-later */
+
+#include <errno.h>
+#include <getopt.h>
+#include <linux/vt.h>
+#include <sys/ioctl.h>
+#include <stdio.h>
+
+#include "alloc-util.h"
+#include "build.h"
+#include "fd-util.h"
+#include "io-util.h"
+#include "log.h"
+#include "main-func.h"
+#include "sd-id128.h"
+#include "sd-journal.h"
+#include "terminal-util.h"
+
+static void help(void) {
+        printf("%s\n\n"
+               "filters the journal to fetch the first message from the current boot with an emergency log level.\n\n"
+               "   -h --help            Show this help\n"
+               "      --version         Show package version\n",
+               program_invocation_short_name);
+}
+
+static int acquire_first_emergency_log_message(char **ret) {
+        sd_journal *j = NULL;
```

Comment:
use `_cleanup_(sd_journal_closep)` here and you can drop the `goto clean_journal` stuff

Reply:
> use `_cleanup_(sd_journal_closep)` here and you can drop the `goto clean_journal` stuff

Awesome.
Thank you sir.